app/experimenter/nimbus-ui/src/components/SummaryTimeline/index.tsx

```diff
@@ -0,0 +1,153 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import ProgressBar from "react-bootstrap/ProgressBar";
+import { NimbusExperimentStatus } from "../../types/globalTypes";
+import pluralize from "../../lib/pluralize";
+import { getExperiment_experimentBySlug } from "../../types/getExperiment";
+
+const humanDate = (date: string): string => {
+  return new Date(date).toLocaleString("en-US", {
+    year: "numeric",
+    month: "short",
+    day: "numeric",
+  });
+};
+
+const experimentStatus = (status: NimbusExperimentStatus) => {
+  return {
+    draft: status === NimbusExperimentStatus.DRAFT,
+    review: status === NimbusExperimentStatus.REVIEW,
+    accepted: status === NimbusExperimentStatus.ACCEPTED,
+    live: status === NimbusExperimentStatus.LIVE,
+    complete: status === NimbusExperimentStatus.COMPLETE,
+  };
+};
+
+const SummaryTimeline = ({
+  experiment,
+}: {
+  experiment: getExperiment_experimentBySlug;
+}) => {
+  const status = experimentStatus(experiment.status!);
+
+  return (
+    <div className="mb-4" data-testid="summary-timeline">
+      <StartEnd
+        {...{
+          status,
+          startDate: experiment.startDate,
+          endDate: experiment.endDate,
+        }}
+      />
+
+      <Progress
+        {...{
+          status,
+          startDate: experiment.startDate,
+          endDate: experiment.endDate,
+        }}
+      />
+
+      <Duration
+        {...{
+          duration: experiment.proposedDuration,
+          enrollment: experiment.proposedEnrollment,
+        }}
+      />
+    </div>
+  );
+};
+
+const StartEnd = ({
+  status,
+  startDate,
+  endDate,
+}: {
+  status: ReturnType<typeof experimentStatus>;
+  startDate: string | null;
+  endDate: string | null;
+}) => (
+  <div className="d-flex">
+    {status.draft || status.review || status.accepted ? (
+      <span className="flex-fill" data-testid="label-not-launched">
+        Not yet launched
+      </span>
+    ) : (
+      <>
+        <span className="flex-fill" data-testid="label-start-date">
+          {humanDate(startDate!)}
+        </span>
+        <span className="flex-fill text-right" data-testid="label-end-date">
+          {humanDate(endDate!)}
+        </span>
+      </>
+    )}
+  </div>
+);
+
+const Progress = ({
+  status,
+  startDate,
+  endDate,
+}: {
+  status: ReturnType<typeof experimentStatus>;
+  startDate: string | null;
+  endDate: string | null;
+}) => {
+  const props: React.ComponentProps<typeof ProgressBar> = {
+    style: { height: 5 },
+    className: "my-2",
+  };
+
+  switch (true) {
```

Comment:
Huh. Haven't used a switch statement this way before. Maybe clearer as if / else?